totp/src/main/java/dev/samstevens/totp/time/NtpTimeProvider.java

```diff
@@ -37,13 +37,19 @@ private NtpTimeProvider(String ntpHostname, int timeout, String dependentClass)
 
     @Override
     public long getTime() throws TimeProviderException {
+        TimeInfo timeInfo;
         try {
-            TimeInfo timeInfo = client.getTime(ntpHost);
-
-            return (long) Math.floor(timeInfo.getReturnTime() / 1000L);
+            timeInfo = client.getTime(ntpHost);
+            timeInfo.computeDetails();
         } catch (Exception e) {
             throw new TimeProviderException("Failed to provide time from NTP server. See nested exception.", e);
         }
+
+        if (timeInfo.getOffset() == null) {
+            throw new TimeProviderException("Failed to calculate NTP offset", null);
```

Comment:
```suggestion
            throw new TimeProviderException("Failed to calculate NTP offset");
```

Since you added the message only constructor can we drop the null here?

Reply:
That was the idea 🤦‍♂️